sky/compositor/paint_context.h

```diff
@@ -13,21 +15,52 @@ namespace compositor {
 
 class PaintContext {
  public:
-  PaintContext(PictureRasterzier& rasterizer,
-               GrContext* gr_context,
-               SkCanvas* canvas);
+  class ScopedFrame {
+   public:
+    PaintContext& paint_context() { return context_; };
+
+    GrContext* gr_context() { return gr_context_; }
+
+    SkCanvas& canvas() { return canvas_; }
+
+    ScopedFrame(ScopedFrame&& frame) = default;
```

Comment:
I'm not sure whether the clang plugin will force you to move this out of line.  I think it might be ok as written.